app/server/controller/src/main/java/io/syndesis/server/controller/integration/online/PublishHandler.java

```diff
@@ -272,15 +271,14 @@ private int countDeployments(IntegrationDeployment deployment) {
      */
     private boolean hasPublishedDeployments(IntegrationDeployment deployment) {
         Integration integration = deployment.getSpec();
-        String id = Labels.sanitize(integration.getId().orElseThrow(() -> new IllegalStateException("Couldn't find the id of the integration")));
+        String id = Labels.validate(integration.getId().orElseThrow(() -> new IllegalStateException("Couldn't find the id of the integration")));
         String version = String.valueOf(integration.getVersion());
 
         Map<String, String> labels = new HashMap<>();
-        labels.put(OpenShiftService.INTEGRATION_ID_LABEL, Labels.sanitize(id));
+        labels.put(OpenShiftService.INTEGRATION_ID_LABEL, Labels.validate(id));
 
         return (int) openShiftService().getDeploymentsByLabel(labels)
             .stream()
-            .filter(d -> !id.equals(d.getMetadata().getLabels().get(OpenShiftService.INTEGRATION_ID_LABEL)))
```

Comment:
why is this not needed here anymore ?

Reply:
because we are querying by label. No need to also filter.